cuqi/problem/_problem.py

```diff
@@ -250,7 +250,7 @@ def MAP(self, disp=True, x0=None) -> CUQIarray:
             print("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
             print("")
 
-        if self._check_posterior((Gaussian, GaussianCov), Gaussian, LinearModel, max_dim=config.MAX_DIM_INV):
+        if self._check_posterior(self, (Gaussian, GaussianCov), Gaussian, LinearModel, max_dim=config.MAX_DIM_INV):
```

Comment:
Just a minor comment here. I am more familiar with using class name to call static methods
```suggestion
        if BayesianProblem._check_posterior(self, (Gaussian, GaussianCov), Gaussian, LinearModel, max_dim=config.MAX_DIM_INV):
```
As static methods are not associated with an object, but with the class. This is my understanding but it could be I'm missing something.

Reply:
I believe that would be a class method.